pkg/kubelet/status/status_manager.go

```diff
@@ -220,16 +220,10 @@ func (m *manager) SetPodStatus(pod *v1.Pod, status v1.PodStatus) {
 	m.updateStatusInternal(pod, status, pod.DeletionTimestamp != nil)
 }
 
-func (m *manager) SetContainerReadiness(podUID types.UID, containerID kubecontainer.ContainerID, ready bool) {
+func (m *manager) SetContainerReadiness(pod *v1.Pod, containerID kubecontainer.ContainerID, ready bool) {
 	m.podStatusesLock.Lock()
 	defer m.podStatusesLock.Unlock()
 
-	pod, ok := m.podManager.GetPodByUID(podUID)
```

Comment:
please let me know if I'm wrong, before this PR all the Pod status handling was centralized and managed by the podManager, with this PR the status manager shortcuts the podmanager?

Reply:
The first thing RemovePod does is remove the pod from PodManager, so the pod is already removed from PodManager when it gets to the status function on a probe status update. The change here is to pipe the pod all the the way through to the status manager so the status update doesn't get ignored.